clientapi/routing/admin.go

```diff
@@ -24,6 +25,86 @@ import (
 	"github.com/matrix-org/dendrite/userapi/api"
 )
 
+func AdminCreateNewToken(req *http.Request) util.JSONResponse {
+	request := struct {
+		Token       string `json:"token"`
+		UsesAllowed int32  `json:"uses_allowed"`
+		ExpiryTime  int64  `json:"expiry_time"`
+		Length      int32  `json:"length"`
+	}{}
+
+	if err := json.NewDecoder(req.Body).Decode(&request); err != nil {
+		return util.JSONResponse{
+			Code: http.StatusBadRequest,
+			JSON: spec.Unknown("Failed to decode request body: " + err.Error()),
+		}
+	}
+	token := request.Token
+	if len(token) > 0 {
```

Comment:
I'd probably remove this and only have something like:

```
if len(token) == 0 {
 // bad request
}
if len(token) > 64 {
// bad request
}
```
or:
```
if len(token) == 0 || len(token) > 64 {
// bad request
}
```
otherwise continue on the happy path and try to validate that the token is valid.